### DISCUSSION TITLE
Documentation regarding `VIRTUAL_ENV` behaviour

### DISCUSSION THREAD

**Discussion by aucampia** [python-poetry/poetry#7335] [Q&A]
As far as I can tell, setting the `VIRTUAL_ENV` environment variable controls where poetry installs to, however I can't find any documentation about this, is there any, and should there not be some?

**Comment by neersighted** [comment]
This is not Poetry-specific behavior, setting `VIRTUAL_ENV` is how virtual environments work. This is technically implementation-defined and not part of [PEP 405](https://peps.python.org/pep-0405/), but it is well-established and stable behavior.

The only 'magic' occurring with this variable is that Poetry will opt out of its own minimal virtual environment management when it is set.

**Comment by chriscarrollsmith** [comment]
If you are seeing `warning: `VIRTUAL_ENV=/some_path/` does not match the project environment path `.venv` and will be ignored`, it means that poetry has detected an in-project virtual environment but has a different one activated in the current shell. To resolve this conflict, you can:

1) Deactivate and remove the poetry-managed environment altogether by typing `poetry env remove`, or
2) Remove the in-project virtual environment by deleting the `.venv` folder
